website/source/docs/providers/aws/r/api_gateway_method_settings.html.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: "aws"
+page_title: "AWS: aws_api_gateway_method_settings"
+sidebar_current: "docs-aws-resource-api-gateway-method-settings"
+description: |-
+  Provides an API Gateway Method Settings, e.g. logging or monitoring.
+---
+
+# aws\_api\_gateway\_method\_settings
+
+Provides an API Gateway Method Settings, e.g. logging or monitoring.
+
+## Example Usage
+
+```hcl
+resource "aws_api_gateway_method_settings" "s" {
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  stage_name  = "${aws_api_gateway_stage.test.stage_name}"
+  method_path = "${aws_api_gateway_resource.test.path_part}/${aws_api_gateway_method.test.http_method}"
```

Comment:
Since it seems like this is just being used to construct a prefix for a longer path anyway (and thus `method_path` is not a first-class concept in the underlying API?), perhaps it would be better to leave these things separated and combine them in code, thus avoiding the need to explain it below as just "concatenate two things together".

Reply:
**path_part** won't work if the resource has a non-root parent though, right? In that case I ended up using **path** which contains the fully qualified path, but the leading slash returned in the path was causing the settings to not be applied properly. My solution for the time being looks like:

```
"${substr("${aws_api_gateway_resource.test.path}/${aws_api_gateway_method.test.http_method}", 1, -1)}"
```